magpie/models.py

```diff
@@ -132,14 +145,309 @@ def __str__(self):
+        status = super(UserStatuses, cls).get(int(status) if str.isnumeric(str(status)) else status)
+        if status:
+            # always convert as enum instance to allow flag combinations (|) and membership compare (in)
+            return UserStatuses(status)
+        return None
+
+    @classmethod
+    def get(cls,
+            status,         # type: Union[None, int, Str, UserStatuses, Iterable[None, int, Str, UserStatuses]]
+            default=None,   # type: Optional[UserStatuses]
+            ):              # type: (...) -> Optional[UserStatuses]
+        """
+        Obtains the combined flag :class:`UserStatuses`
+        """
+        if status is None:
+            return default
+        if status in ["all", "All", "ALL"]:
+            return cls.all()
+        if isinstance(status, str) and "," in status:
+            status = status.split(",")
+        if isinstance(status, (str, int)):
+            return cls._get_one(status)
+        combined = None
+        for _status in status:
+            _status = cls._get_one(_status)
+            if combined is not None and _status is not None:
+                combined = (combined | _status)
+            else:
+                combined = combined or _status
+        return UserStatuses(combined)
+
+    @classmethod
+    def allowed(cls):
+        # type: () -> List[Union[None, int, Str]]
+        """
+        Returns all supported representation values that can be mapped to a valid status for :class:`UserSearchService`.
+        """
+        allowed = cls.values()  # literal int
+        allowed.extend(list(str(status) for status in allowed))  # by str repr of int value
+        names = cls.names()
+        allowed.extend(names)  # by literal name of status
+        allowed.extend([name.upper() for name in names])
+        allowed.extend([name.lower() for name in names])
+        allowed.extend([None, "all", "All", "ALL"])  # unspecified (valid users omit pending) or literally 'all' users
+        return allowed
+
+    @classmethod
+    def all(cls):
+        """
+        Representation of all flags combined.
+        """
+        return UserStatuses(sum(cls.values()))
+
+    # nothing to do, only improve typing to avoid complaints about expecting 'int'
+    def __or__(self, other):  # type: (Union[UserStatuses, int]) -> UserStatuses
+        return super(UserStatuses, self).__or__(other)
+
+    # nothing to do, only improve typing to avoid complaints about expecting 'int'
+    def __and__(self, other):  # type: (Union[UserStatuses, int]) -> UserStatuses
+        return super(UserStatuses, self).__and__(other)
+
+    # nothing to do, only improve typing to avoid complaints about expecting 'int'
+    def __xor__(self, other):  # type: (Union[UserStatuses, int]) -> UserStatuses
+        return super(UserStatuses, self).__xor__(other)
+
+    def __iter__(self):
+        values = decompose_enum_flags(self)
+        return iter(values)
+
+    def __len__(self):
```

Comment:
Is it the number of flags? Are you sure about the implementation?

Reply:
What is the possible values of self.value? Can it be 6 for pending & webhook_error?

Reply:
This code snippet was taken from somewhere in the enum code and related forums. 
There are 2 possible interpretation of `__len__`, depending if the full enum class or a member is processed. 
When full enum, it should return the number of possible distinct flags (3 so far in this case). 
When a member is passed, it must return the number of set flags, which can be a combination of members. 


```python
>>> from magpie.models import UserStatuses
>>> len(UserStatuses)
3
>>> len(UserStatuses.OK)
1
>>> len(UserStatuses.OK|UserStatuses.WebhookError)
2
>>> UserStatuses.WebhookError + UserStatuses.Pending
6
>>> len(UserStatuses.WebhookError|UserStatuses.Pending)
2
```

There are some tests that validate the specific behaviour of `UserStatuses`.

Reply:
Ok, the loop is pretty genious!

Reply:
I definitely have no credit. Stack Overflow is your friend :)